### PR TITLE
DBAAS-738: avoid adopting DB instances in clusters

### PR DESCRIPTION
**Pull request by xieshenzh**
When adopting DB instances, skip the DB instances that are in clusters.

### CODE REVIEW

**Review by jeremyary — APPROVED** [review]

**Review by redhatHameed — APPROVED** [review]
LGTM

File: controllers/rdsinventory_controller.go [hunk 1]
```diff
@@ -460,6 +463,9 @@ func (r *RDSInventoryReconciler) Reconcile(ctx context.Context, req ctrl.Request
 		waitForAdoptedResource := false
 		for i := range adoptedDBInstanceList.Items {
```
Comment (jianrongzhang89) [inline controllers/rdsinventory_controller.go:464]: This loop can be simplified to:
for _, adoptedDBInstance := range adoptedDBInstanceList.Items { ... }

Comment (xieshenzh) [reply controllers/rdsinventory_controller.go:464]: The code at line 485 takes the address of the loop variable. 
So if I do what you suggest, there will be a warning of `G601: Implicit memory aliasing in for loop`.

Comment (jianrongzhang89) [reply controllers/rdsinventory_controller.go:464]: OK

**Review by jianrongzhang89 — APPROVED** [review]
LGTM

**Pull request merged**